google-cloud-spanner/src/test/java/com/google/cloud/spanner/MockSpannerServiceImpl.java

```diff
@@ -596,7 +596,7 @@ private static void checkStreamException(
   private int maxNumSessionsInOneBatch = 100;
   private int maxTotalSessions = Integer.MAX_VALUE;
   private AtomicInteger numSessionsCreated = new AtomicInteger();
-
+  private boolean onCommitThrowSessionNotFoundException = false;
```

Comment:
Could we do the following instead of adding yet another field to this class:


1. Extract the creation of a `SessionNotFound` error from `setSessionNotFound` into a public method, so something like this:

```java
  public StatusRuntimeException createSessionNotFoundException(String name) {
    ResourceInfo resourceInfo =
        ResourceInfo.newBuilder()
            .setResourceType(SpannerExceptionFactory.SESSION_RESOURCE_TYPE)
            .setResourceName(name)
            .build();
    Metadata.Key<ResourceInfo> key =
        Metadata.Key.of(
            resourceInfo.getDescriptorForType().getFullName() + Metadata.BINARY_HEADER_SUFFIX,
            ProtoLiteUtils.metadataMarshaller(resourceInfo));
    Metadata trailers = new Metadata();
    trailers.put(key, resourceInfo);
    return
        Status.NOT_FOUND
            .withDescription(String.format("Session not found: Session with id %s not found", name))
            .asRuntimeException(trailers);
  }
```

2. Re-use the above method in the `setSessionNotFound` method.
3. Use the following in the test method(s) that use this:

```java    mockSpanner.setCommitExecutionTime(SimulatedExecutionTime.ofException(mockSpanner.createSessionNotFoundException("test-session-name")));
```